deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -0,0 +1,932 @@
+	mcmslib "github.com/smartcontractkit/mcms"
+	mcmssdk "github.com/smartcontractkit/mcms/sdk"
+	mcmstypes "github.com/smartcontractkit/mcms/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset/globals"
+	commoncs "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	commontypes "github.com/smartcontractkit/chainlink/deployment/common/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/v1_6_0/fee_quoter"
+)
+
+/*
+Future improvements:
+- Enable connecting directly to the prod router through PromoteNewChainForTestingChangeset
+- Align configuration structs with whatever is simplest for BIX team to use
+- Add more validation coverage to the precondition functions
+*/
+
+var (
+	// AddCandidatesForNewChainChangeset deploys a new chain and adds its exec and commit plugins as candidates on the home chain.
+	AddCandidatesForNewChainChangeset = deployment.CreateChangeSet(addCandidatesForNewChainLogic, addCandidatesForNewChainPrecondition)
+	// PromoteNewChainForTestingChangeset promotes exec and commit plugin candidates for the new chain on the home chain.
+	// It also connects the new chain to various destination chains through the test router.
+	// This changeset should be run after AddCandidatesForNewChainChangeset.
+	PromoteNewChainForTestingChangeset = deployment.CreateChangeSet(promoteNewChainForTestingLogic, promoteNewChainForTestingPrecondition)
+	// ConnectNewChainChangeset activates connects a new chain with other chains by updating onRamp, offRamp, and router contracts.
+	// If connecting to production routers, you should have already run PromoteNewChainForTestingChangeset.
+	ConnectNewChainChangeset = deployment.CreateChangeSet(connectNewChainLogic, connectNewChainPrecondition)
+)
+
+///////////////////////////////////
+// START AddCandidatesForNewChainChangeset
+///////////////////////////////////
+
+// ChainDefinition defines how a chain should be configured on both remote chains and itself.
+type ChainDefinition struct {
+	ConnectionConfig
+	// Selector is the chain selector of this chain.
+	Selector uint64
+	// GasPrice defines the USD price (18 decimals) per unit gas for this chain as a destination.
+	GasPrice *big.Int
+	// TokenPrices define the USD price (18 decimals) per 1e18 of the smallest token denomination for various tokens on this chain.
+	TokenPrices map[common.Address]*big.Int
+	// FeeQuoterDestChainConfig is the configuration on a fee quoter for this chain as a destination.
+	FeeQuoterDestChainConfig fee_quoter.FeeQuoterDestChainConfig
+}
+
+// NewChainDefinition defines how a NEW chain should be configured.
+type NewChainDefinition struct {
+	ChainDefinition
```

Comment:
is there a need to support multiple chains at the same time here?

Reply:
Not at this time, 1 new chain at a time is what is in scope here. BIX can iterate later as needed.

Reply:
Okay. This might help with multiple parallel chain addition requirement without worrying about donid